streaming/src/test/java/org/apache/spark/streaming/JavaAPISuite.java

```diff
@@ -1707,6 +1708,71 @@ public Integer call(String s) throws Exception {
     Utils.deleteRecursively(tempDir);
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testContextGetOrCreate() throws InterruptedException {
+
+    final SparkConf conf = new SparkConf()
+        .setMaster("local[2]")
+        .setAppName("test")
+        .set("newContext", "true");
+
+    File emptyDir = Files.createTempDir();
+    emptyDir.deleteOnExit();
+    StreamingContextSuite contextSuite = new StreamingContextSuite();
+    String corruptedCheckpointDir = contextSuite.createCorruptedCheckpoint();
+    String checkpointDir = contextSuite.createValidCheckpoint();
+
+    // Function to create JavaStreamingContext without any output operations
+    // (used to detect the new context)
+    Function0<JavaStreamingContext> creatingFunc = new Function0<JavaStreamingContext>() {
+      public JavaStreamingContext call() {
+        return new JavaStreamingContext(conf, Seconds.apply(1));
+      }
+    };
+
+    ssc = JavaStreamingContext.getOrCreate(emptyDir.getAbsolutePath(), creatingFunc);
+    Assert.assertTrue("new context not created",
```

Comment:
Is your goal here to assert that `creatingFunc` was not called?  This seems like kind of a roundabout way of doing that.  I know that Mockito makes it pretty easy to write assertions that check the number of times that methods are invoked; alternatively, I guess you could just stick a "number of times called" counter in your function and check it in your assert.